pkg/sql/resolver.go

```diff
@@ -69,77 +44,94 @@ func (p *planner) ResolveUncachedDatabaseByName(
-// schema. If explicitPrefix is set, the returned table names will have an
-// explicit schema and catalog name.
-func GetObjectNames(
-	ctx context.Context,
-	txn *kv.Txn,
-	sc SchemaResolver,
-	codec keys.SQLCodec,
-	dbDesc *DatabaseDescriptor,
-	scName string,
-	explicitPrefix bool,
-) (res TableNames, err error) {
-	return sc.LogicalSchemaAccessor().GetObjectNames(ctx, txn, codec, dbDesc, scName,
-		tree.DatabaseListFlags{
-			CommonLookupFlags: sc.CommonLookupFlags(true /* required */),
-			ExplicitPrefix:    explicitPrefix,
-		})
-}
-
-// ResolveExistingTableObject looks up an existing object.
-// If required is true, an error is returned if the object does not exist.
-// Optionally, if a desired descriptor type is specified, that type is checked.
+// runWithOptions sets the provided resolution flags for the
+// duration of the call of the passed argument fn.
 //
-// The object name is modified in-place with the result of the name
-// resolution, if successful. It is not modified in case of error or
-// if no object is found.
-func ResolveExistingTableObject(
-	ctx context.Context,
-	sc SchemaResolver,
-	tn *TableName,
-	lookupFlags tree.ObjectLookupFlags,
-	requiredType ResolveRequiredType,
-) (res *ImmutableTableDescriptor, err error) {
-	// TODO: As part of work for #34240, an UnresolvedObjectName should be
-	//  passed as an argument to this function.
-	un := tn.ToUnresolvedObjectName()
-	desc, prefix, err := resolveExistingObjectImpl(ctx, sc, un, lookupFlags, requiredType)
-	if err != nil || desc == nil {
-		return nil, err
+// This is meant to be used like this (for example):
+//
+// var someVar T
+// var err error
+// p.runWithOptions(resolveFlags{skipCache: true}, func() {
+//    someVar, err = ResolveExistingTableObject(ctx, p, ...)
+// })
+// if err != nil { ... }
+// use(someVar)
+func (p *planner) runWithOptions(flags resolveFlags, fn func()) {
+	if flags.skipCache {
+		defer func(prev bool) { p.avoidCachedDescriptors = prev }(p.avoidCachedDescriptors)
+		p.avoidCachedDescriptors = true
 	}
-	tn.ObjectNamePrefix = prefix
-	return desc.(*ImmutableTableDescriptor), nil
+	fn()
 }
 
-// ResolveMutableExistingTableObject looks up an existing mutable object.
-// If required is true, an error is returned if the object does not exist.
-// Optionally, if a desired descriptor type is specified, that type is checked.
-//
-// The object name is modified in-place with the result of the name
-// resolution, if successful. It is not modified in case of error or
-// if no object is found.
-func ResolveMutableExistingTableObject(
-	ctx context.Context,
-	sc SchemaResolver,
-	tn *TableName,
-	required bool,
-	requiredType ResolveRequiredType,
-) (res *MutableTableDescriptor, err error) {
-	lookupFlags := tree.ObjectLookupFlags{
-		CommonLookupFlags: tree.CommonLookupFlags{Required: required},
-		RequireMutable:    true,
+type resolveFlags struct {
+	skipCache bool
+}
+
+func (p *planner) ResolveMutableTableDescriptor(
```

Comment:
You may want to keep comment docstrings here and below. (commit "sql/catalog/resolver: extract resolver...")

Reply:
not sure I follow, what comments did I remove?

Reply:
The function is exported, so a comment is warranted